src/core/Akka/IO/TcpOutgoingConnection.cs

```diff
@@ -73,12 +74,14 @@ protected override void PreStart()
                     var resolved = Dns.ResolveName(remoteAddress.Host, Context.System, Self);
                     if (resolved == null)
                         Become(Resolving(remoteAddress));
-                    else
-                        Register(new IPEndPoint(resolved.Addr, remoteAddress.Port));
+                    else if(resolved.Ipv4.Any() && resolved.Ipv6.Any()) // one of both families
+                        Register(new IPEndPoint(resolved.Ipv4.FirstOrDefault(), remoteAddress.Port), new IPEndPoint(resolved.Ipv6.FirstOrDefault(), remoteAddress.Port));
```

Comment:
Added a fallback scheme here that only gets invoked when using DNS endpoints. If using DNS, first attempt to connect to a socket is done with IPV4 (default from earlier versions of Akka.IO)... If that fails, second attempt will swap and use IPV6. Third attempt will swap back to IPV4. Fourth will swap back to IPV6... Etc... Idea is to provide some help given the ambiguity of DNS resolution and the address family being used.

As I noted here https://github.com/akkadotnet/akka.net/issues/3064, this could be improved to use the explicit `AddressFamily` provided on the `DnsEndPoint` when it's available. We could also extend this in the future to support falling back through multiple resolved addresses until the right one is found. But I decided to leave well-enough alone for now.